src/Data/Expr.purs

```diff
@@ -4,259 +4,157 @@ import Prelude
 
 import Data.Argonaut (class DecodeJson, class EncodeJson, Json, decodeJson, encodeJson, jsonEmptyObject, (.:), (:=), (~>))
 import Data.Either (Either(..))
-import Data.Leibniz (type (~), coerceSymm)
-import Type.Prelude (Proxy(..))
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Maybe (Maybe, maybe)
+
+type Key = String
 
 data ExprType
   = Boolean Boolean
   | Int Int
   | String String
 
+derive instance genericExprType :: Generic ExprType _
+
+instance showExprType :: Show ExprType where
+  show = genericShow
+
+instance encodeJsonExprType :: EncodeJson ExprType where
+  encodeJson = case _ of
+    Boolean x -> encodeJson x
+    Int x -> encodeJson x
+    String x -> encodeJson x
+
 reflectType :: ExprType -> String
 reflectType (Boolean _) = "Boolean"
 reflectType (Int _) = "Int"
 reflectType (String _) = "String"
 
-class (Show a, Eq a) <= Expressible a where
-  toExprType :: a -> ExprType
-  reflectProxy :: Proxy a -> String
-  print :: a -> String
-
-instance exprBoolean :: Expressible Boolean where
-  toExprType = Boolean
-  reflectProxy _ = "Boolean"
-  print = show
+print :: ExprType -> ExprType
```

Comment:
I think this is typed not right, it was thrown together when adding `Print` back in the mix, but it looks like we probably want the type to be `ExprType -> String` and we can do the wrapping where it's used.